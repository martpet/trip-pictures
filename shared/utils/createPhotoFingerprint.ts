import { PhotoFingerprint } from '../types';

export function createPhotoFingerprint({
  date,
  lon,
  lat,
}: {
  date: string;
  lon?: number;
  lat?: number;
}): PhotoFingerprint {
  return `${date}_${Number(lon)}_${Number(lat)}`;
}
