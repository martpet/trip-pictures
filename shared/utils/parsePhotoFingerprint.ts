import { PhotoFingerprint } from '../types';

export function parsePhotoFingerprint(fingerprint: PhotoFingerprint) {
  const [date, lon, lat] = fingerprint.split('_');

  return {
    date,
    lon: Number(lon),
    lat: Number(lat),
  };
}
