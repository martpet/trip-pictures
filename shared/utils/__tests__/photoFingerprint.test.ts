import { createPhotoFingerprint } from '../createPhotoFingerprint';
import { parsePhotoFingerprint } from '../parsePhotoFingerprint';

describe('createPhotoFingerprint', () => {
  it('returns a correct value', () => {
    expect(
      createPhotoFingerprint({
        date: 'dummyDate',
        lon: 1.23,
        lat: 3.45,
      })
    ).toMatchSnapshot();
  });
});

describe('parsePhotoFingerprint', () => {
  it('returns a correct value', () => {
    expect(
      parsePhotoFingerprint('2022-08-13T09:52:30_20.176583_39.505653')
    ).toMatchSnapshot();
  });
});
