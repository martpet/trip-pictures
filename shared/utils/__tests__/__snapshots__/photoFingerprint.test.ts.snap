// Vitest Snapshot v1

exports[`createPhotoFingerprint > returns a correct value 1`] = `"dummyDate_1.23_3.45"`;

exports[`parsePhotoFingerprint > returns a correct value 1`] = `
{
  "date": "2022-08-13T09:52:30",
  "lat": 39.505653,
  "lon": 20.176583,
}
`;
